vllm/entrypoints/llm.py

```diff
@@ -251,4 +256,23 @@ def _run_engine(self, use_tqdm: bool) -> List[RequestOutput]:
         # This is necessary because some requests may be finished earlier than
         # its previous requests.
         outputs = sorted(outputs, key=lambda x: int(x.request_id))
-        return outputs
\ No newline at end of file
+        return outputs
+
+    def _validate_prompt(
+        self,
+        prompt: Optional[str] = None,
+        prompt_token_ids: Optional[List[int]] = None,
+        truncate_prompt_tokens: Optional[Annotated[int, Field(ge=1)]] = None
+    ) -> Tuple[Optional[str], Optional[List[int]]]:
```

Comment:
This function re-implements the same logic as [this function](https://github.com/vllm-project/vllm/blob/0650e5935b0f6af35fb2acf71769982c47b804d7/vllm/entrypoints/openai/serving_engine.py#L188-L194). It's worth thinking about whether we should have it implemented in a single place and re-used accordingly. I think that would be preferable, unless there is some strong reason to have it duplicated.

Reply:
I'm also studying #3512 to understand if that needs to be merged before this one.

Reply:
yes this is a good point. let me know what do you think is the best and I'll implement it

Reply:
For now let's just make sure that the behaviour is equivalent (see the below discussion re: using truncation from the tokenizer). I think it would make more sense to think about refactoring it after #3512 is merged because after that point the completions API will be using the tokenizer from the engine anyway.